// nylas/lib/esm/models/webhooks.js
/**
 * Enum representing the available webhook triggers.
 */
export var WebhookTriggers;
(function (WebhookTriggers) {
    // Calendar triggers
    WebhookTriggers["CalendarCreated"] = "calendar.created";
    WebhookTriggers["CalendarUpdated"] = "calendar.updated";
    WebhookTriggers["CalendarDeleted"] = "calendar.deleted";
    // Event triggers
    WebhookTriggers["EventCreated"] = "event.created";
    WebhookTriggers["EventUpdated"] = "event.updated";
    WebhookTriggers["EventDeleted"] = "event.deleted";
    // Grant triggers
    WebhookTriggers["GrantCreated"] = "grant.created";
    WebhookTriggers["GrantUpdated"] = "grant.updated";
    WebhookTriggers["GrantDeleted"] = "grant.deleted";
    WebhookTriggers["GrantExpired"] = "grant.expired";
    // Message triggers
    WebhookTriggers["MessageCreated"] = "message.created";
    WebhookTriggers["MessageUpdated"] = "message.updated";
    WebhookTriggers["MessageSendSuccess"] = "message.send_success";
    WebhookTriggers["MessageSendFailed"] = "message.send_failed";
    WebhookTriggers["MessageBounceDetected"] = "message.bounce_detected";
    // Message tracking triggers
    WebhookTriggers["MessageOpened"] = "message.opened";
    WebhookTriggers["MessageLinkClicked"] = "message.link_clicked";
    WebhookTriggers["ThreadReplied"] = "thread.replied";
    // ExtractAI triggers
    WebhookTriggers["MessageIntelligenceOrder"] = "message.intelligence.order";
    WebhookTriggers["MessageIntelligenceTracking"] = "message.intelligence.tracking";
    // Folder triggers
    WebhookTriggers["FolderCreated"] = "folder.created";
    WebhookTriggers["FolderUpdated"] = "folder.updated";
    WebhookTriggers["FolderDeleted"] = "folder.deleted";
    // Contact triggers
    WebhookTriggers["ContactUpdated"] = "contact.updated";
    WebhookTriggers["ContactDeleted"] = "contact.deleted";
    // Scheduler triggers
    WebhookTriggers["BookingCreated"] = "booking.created";
    WebhookTriggers["BookingPending"] = "booking.pending";
    WebhookTriggers["BookingRescheduled"] = "booking.rescheduled";
    WebhookTriggers["BookingCancelled"] = "booking.cancelled";
    WebhookTriggers["BookingReminder"] = "booking.reminder";
})(WebhookTriggers || (WebhookTriggers = {}));
