// nylas/lib/esm/resources/configurations.js
import { Resource } from './resource.js';
export class Configurations extends Resource {
    /**
     * Return all Configurations
     * @return A list of configurations
     */
    list({ identifier, overrides, }) {
        return super._list({
            overrides,
            path: `/v3/grants/${identifier}/scheduling/configurations`,
        });
    }
    /**
     * Return a Configuration
     * @return The configuration
     */
    find({ identifier, configurationId, overrides, }) {
        return super._find({
            path: `/v3/grants/${identifier}/scheduling/configurations/${configurationId}`,
            overrides,
        });
    }
    /**
     * Create a Configuration
     * @return The created configuration
     */
    create({ identifier, requestBody, overrides, }) {
        return super._create({
            path: `/v3/grants/${identifier}/scheduling/configurations`,
            requestBody,
            overrides,
        });
    }
    /**
     * Update a Configuration
     * @return The updated Configuration
     */
    update({ configurationId, identifier, requestBody, overrides, }) {
        return super._update({
            path: `/v3/grants/${identifier}/scheduling/configurations/${configurationId}`,
            requestBody,
            overrides,
        });
    }
    /**
     * Delete a Configuration
     * @return The deleted Configuration
     */
    destroy({ identifier, configurationId, overrides, }) {
        return super._destroy({
            path: `/v3/grants/${identifier}/scheduling/configurations/${configurationId}`,
            overrides,
        });
    }
}
